driver-core/src/main/com/mongodb/connection/DefaultClusterableServerFactory.java

```diff
@@ -20,67 +20,64 @@
     private final ConnectionPoolSettings connectionPoolSettings;
     private final StreamFactory streamFactory;
     private final List<MongoCredential> credentialList;
     private final ConnectionPoolListener connectionPoolListener;
-    private final ConnectionListener connectionListener;
+    private final ServerMonitorListener serverMonitorListener;
     private final StreamFactory heartbeatStreamFactory;
     private final CommandListener commandListener;
     private final String applicationName;
     private final MongoDriverInformation mongoDriverInformation;
 
-    DefaultClusterableServerFactory(final ClusterId clusterId, final ClusterSettings clusterSettings, final ServerSettings settings,
+    DefaultClusterableServerFactory(final ClusterId clusterId, final ClusterSettings clusterSettings, final ServerSettings serverSettings,
                                     final ConnectionPoolSettings connectionPoolSettings, final StreamFactory streamFactory,
                                     final StreamFactory heartbeatStreamFactory, final List<MongoCredential> credentialList,
-                                    final ConnectionListener connectionListener, final ConnectionPoolListener connectionPoolListener,
+                                    final ConnectionPoolListener connectionPoolListener, final ServerMonitorListener serverMonitorListener,
                                     final CommandListener commandListener, final String applicationName,
                                     final MongoDriverInformation mongoDriverInformation) {
         this.clusterId = clusterId;
         this.clusterSettings = clusterSettings;
-        this.settings = settings;
+        this.serverSettings = serverSettings;
         this.connectionPoolSettings = connectionPoolSettings;
         this.streamFactory = streamFactory;
         this.credentialList = credentialList;
         this.connectionPoolListener = connectionPoolListener;
-        this.connectionListener = connectionListener;
+        this.serverMonitorListener = serverMonitorListener;
         this.heartbeatStreamFactory = heartbeatStreamFactory;
         this.commandListener = commandListener;
         this.applicationName = applicationName;
         this.mongoDriverInformation = mongoDriverInformation;
     }
 
     @Override
-    public ClusterableServer create(final ServerAddress serverAddress, final ServerListener serverListener) {
+    public ClusterableServer create(final ServerAddress serverAddress, final ServerListener initialServerListener) {
         ConnectionPool connectionPool = new DefaultConnectionPool(new ServerId(clusterId, serverAddress),
-                new InternalStreamConnectionFactory(streamFactory, credentialList, connectionListener, applicationName,
+                new InternalStreamConnectionFactory(streamFactory, credentialList, applicationName,
                         mongoDriverInformation), connectionPoolSettings, connectionPoolListener);
```

Comment:
Similar to other comments, let's stop sending both connectionPoolSettings and connectionPoolListener to the DefaultConnectionPool